src/net.c

```diff
@@ -991,6 +991,13 @@ static int hyper_setup_interface(struct rtnl_handle *rth,
 	req.n.nlmsg_type = RTM_NEWADDR;
 	req.ifa.ifa_family = AF_INET;
 
+	if (iface->device) {
```

Comment:
So now we have the following checks on iface->device:

````
if (iface->device) {
     wait_for_nic(iface->device)
         if (!iface->device)
             return -1
````

It seems we could write nicer checks in general. It feels a bit like we end up checking the same pointer at every function call. In this case, maybe adding saying that wait_for_nic() accepts a NULL as a valid value (saying: iface->device is optional, waiting on NULL device name is just a no-op) could make the code a bit nicer I think.

This is similar to how go treats nil is some cases. nil is a valid input for `len()`, it just returns 0 instead of an error and that allows removing some error paths.

Just a general comment, you don't need to address it.